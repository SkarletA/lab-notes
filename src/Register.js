import React from "react";

export function Login() {
  return (
    <section className="register">
      <div className="popup">
        <div className="title">
          <p>Create an account</p>
          <h3>BlogSks</h3>
          <p>it is easy and simple</p>
        </div>
        <form className="form-register">
          <input className="input-email" type="email" placeholder="Email: " />
          <input
            className="input-password"
            type="password"
            placeholder="Password: "
          />
          <button>Register</button>
        </form>
      </div>
    </section>
  );
}
