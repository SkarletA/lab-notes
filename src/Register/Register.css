
.register {
  align-items: center;
  background: rgba(32,35,41,.8);
  bottom: -10px;
  color: white;
  display: flex;
  justify-content: center;
  left: -10px;
  position: fixed;
  right: -10px;
  top: -10px;
}
.popup{
  align-items: center;
  background-color: #FFE3D8;
  border-radius: 1em;
  border: 0.1em solid #7D1935;
  display: flex;
  flex-direction: column;
  height: 25em;
  justify-content: center;
  width: 20em;
}
.btn-close {
  background-color: #FFE3D8;
  border: none;
  bottom: 1em;
  color:#7D1935;
  font-size: 1.2em;
  left: 6em;
  position: relative;
  right: 0;
}

.register button:hover {
  color: blue;
}

h3, p {
  align-items: center;
  color:#7D1935;
  display: flex;
  flex-direction: column;
  margin-block-start: 0;
  margin-block-end: 0;
  margin:0.5em;
}
.input-email, .input-password {
  align-items: center;
  background-color: white;
  border: 0.2em solid #7D1935;
  border-radius: 0.2em;
  color: rgba(0, 0, 0, 0.5);
  cursor: pointer;
  display: flex;
  height: 3em;
  margin-bottom: 1em;
  padding-left: 0.5em;
  width: 15em;
}
.red{
  color:red;
  margin-top: 0.1em;
  margin-bottom: 0.1em;
  display: flex;
  justify-content: center;
}
.btn-register {
  background-color: #7D1935;
  border: 0.2em solid white;
  border-radius: 0.8em;
  color: white;
  cursor: pointer;
  height: 3em;
  width: 15em;
}