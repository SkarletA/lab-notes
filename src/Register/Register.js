import React from "react";
import './Register.css';

export function Register({openModalRegister, closeModalRegister}){
  if (!openModalRegister) return null
  return (
    <section className ="register">
      <div className ="popup">
      <button onClick={closeModalRegister} className ='btn-close'>X</button>
        <div className ="title">
          <p>Create an account</p>
          <h3>BlogSks</h3>
          <p>it is easy and simple</p>
        </div>
        <form className="form-register">
          <input className="input-email" type="email" placeholder="Email: " />
          <input
            className="input-password"
            type="password"
            placeholder="Password: "
          />
          <button className="btn-register">Register</button>
        </form>
      </div>
    </section>
  );
}
