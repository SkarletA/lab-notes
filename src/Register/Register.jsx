import React, {useState} from "react";
import { useNavigate } from "react-router-dom";
import './Register.css';

import "../Firebase/firebase";
import { useCreateUserWithEmailAndPassword } from 'react-firebase-hooks/auth';
import { getAuth } from "firebase/auth";
import { app } from "../Firebase/firebaseconfig";

export function Register({openModalRegister, closeModalRegister}){
  const navigate = useNavigate();
  
  const auth = getAuth(app);
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [
    createUserWithEmailAndPassword,
    user,
    loading,
    error,
  ] = useCreateUserWithEmailAndPassword(auth);
  
  if (error) {
    return (
      <div>
        <p>Error: {error.message}</p>
      </div>
    );
  }
  if (loading) {
    return <p>Loading...</p>;
  }
  if (user) {
    return (
      <div>
        <p>Registered User: {user.email}</p>
      </div>
    );
  }

  if (!openModalRegister) return null

  const submitRegistration = async (e) => {
    e.preventDefault();
    const expEmail = /^\w+([.+-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,4})+$/;
    const expPassword = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[$@$!%*?&])([A-Za-z\d$@$!%*?&]|[^ ]){8,15}$/;
    if (expEmail.test(email) && expPassword.test(password)) {
      const user = await createUserWithEmailAndPassword(email, password);
      navigate('/blog');
      console.log(user);
    } else {
      const alertEmailR = document.getElementById('alertEmailR');
      alertEmailR.innerHTML = '<span class="red"> Email or password invalid </span>';
    }
    
  }
  return (
    <section className ="register">
      <div className ="popup">
      <button onClick={closeModalRegister} className ='btn-close'>X</button>
        <div className ="title">
          <p>Create an account</p>
          <h3>BlogSks</h3>
          <p>it is easy and simple</p>
        </div>
        <form className="form-register">
          <input className="input-email" 
          id="emailR" 
          type="email" 
          placeholder="Email: " 
          onChange={(e)=> setEmail(e.target.value)}/>
          <input
            className="input-password"
            id="passwordR"
            type="password"
            placeholder="Password: "
            onChange={(e)=> setPassword(e.target.value)}
          />
          <span id="alertEmailR" className="red"></span>
          <button className="btn-register" 
          onClick={submitRegistration}
          >Register</button>
        </form>
      </div>
    </section>
  );
}
