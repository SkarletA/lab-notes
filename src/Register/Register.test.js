/* eslint-disable no-shadow */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Register } from './Register';

describe(('Test component Register'), () => {
  test('render component Register', () => {
    let register;
    act(() => {
      register = create(
        <Router>
          <Register />
        </Router>,
      );
    });
    expect(register.toJSON()).toMatchSnapshot();
  });
});
