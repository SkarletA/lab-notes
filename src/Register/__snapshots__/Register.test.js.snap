// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Register render component Register 1`] = `null`;
